modules/consensus/consensusset.go

```diff
@@ -141,9 +143,19 @@ func New(gateway modules.Gateway, persistDir string) (*ConsensusSet, error) {
 		// Register RPCs
 		gateway.RegisterRPC("SendBlocks", cs.rpcSendBlocks)
 		gateway.RegisterRPC("RelayBlock", cs.rpcRelayBlock) // COMPATv0.5.1
-		gateway.RegisterRPC("RelayHeader", cs.rpcRelayHeader)
+		gateway.RegisterRPC("RelayHeader", cs.threadedRPCRelayHeader)
 		gateway.RegisterRPC("SendBlk", cs.rpcSendBlk)
 		gateway.RegisterConnectCall("SendBlocks", cs.threadedReceiveBlocks)
+		cs.tg.OnStop(func() {
+			cs.mu.Lock()
+			defer cs.mu.Unlock()
```

Comment:
I'm having trouble reasoning about whether this should be `OnStop` or `AfterStop`. What exactly happens when the RPCs get removed? Does that just mean that incoming connections are blocked? It shouldn't affect anything else right? Or would calling it early somehow disrupt an ongoing network call?

If we call `OnStop` instead of `AfterStop`, it means the de-registration will happen while there are RPCs in progress. Is that okay?